nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/OAuth2Object.java

```diff
@@ -0,0 +1,165 @@
+        result.put("authorizationURL", provider.getClientId() == null ? null : provider.getAuthorizationUrl(request));
+        result.put("authorizationServerURL", provider.getAuthorizationServerURL());
+        result.put("tokenServerURL", provider.getTokenServerURL());
+
+        String username = request.getUserPrincipal().getName();
+        NuxeoOAuth2Token token = getToken(provider, username);
+        boolean hasToken = (token != null);
+        String login = hasToken ? token.getServiceLogin() : null;
+        result.put("hasToken", hasToken);
+        result.put("userId", login);
+
+        return buildResponse(Status.OK, result);
+    }
+
+    /**
+     * Retrieves a valid access token for a given provider and the current user.
+     * If expired, the token will be refreshed.
+     */
+    @GET
+    @Path("provider/{providerId}/token")
+    public Response getToken(@PathParam("providerId") String providerId,
+                             @Context HttpServletRequest request) throws IOException, RestOperationException {
+
+        NuxeoOAuth2ServiceProvider provider = getProvider(providerId);
+
+        String username = request.getUserPrincipal().getName();
+        NuxeoOAuth2Token token = getToken(provider, username);
+        if (token == null) {
+            return Response.status(Status.NOT_FOUND).build();
+        }
+        Credential credential = getCredential(provider, token);
+
+        if (credential == null) {
+            return Response.status(Status.NOT_FOUND).build();
+        }
+        Long expiresInSeconds = credential.getExpiresInSeconds();
+        if (expiresInSeconds != null && expiresInSeconds <= 0) {
+            credential.refreshToken();
+        }
+        Map<String,Object> result = new HashMap<>();
+        result.put("token", credential.getAccessToken());
+        return buildResponse(Status.OK, result);
+
+    }
+
+    private NuxeoOAuth2Token getToken(NuxeoOAuth2ServiceProvider provider, String nxuser) {
+        Map<String, Serializable> filter = new HashMap<>();
+        filter.put("serviceName", provider.getId());
+        filter.put(NuxeoOAuth2Token.KEY_NUXEO_LOGIN, nxuser);
+        return Framework.doPrivileged(() -> {
+            List<DocumentModel> entries = provider.getCredentialDataStore().query(filter);
+            if (entries != null) {
+                if (entries.size() > 1) {
+                    throw new NuxeoException("Found multiple " + provider.getId() + " accounts for " + nxuser);
+                } else if (entries.size() == 1) {
+                    return new NuxeoOAuth2Token(entries.get(0));
+                }
+            }
+            return null;
+        });
+    }
+
+    private Credential getCredential(NuxeoOAuth2ServiceProvider provider, NuxeoOAuth2Token token) {
+        return provider.loadCredential(
+            (provider instanceof AbstractOAuth2UserEmailProvider) ? token.getServiceLogin() : token.getNuxeoLogin());
```

Comment:
By default, the nuxeo login is used: https://github.com/nuxeo/nuxeo/blob/master/nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/providers/NuxeoOAuth2ServiceProvider.java#L152-L156